app/controllers/responses_controller.rb

```diff
@@ -0,0 +1,43 @@
+class ResponsesController < ApplicationController
+
+def new
+  if params[:answer_id]
+    @responsible = "Answer"
```

Comment:
You may be able to shorten this by examining the url to figure out which route you are in.